### PR TITLE
Use "net" instead of "sc" to restart minion service

### LINKED ISSUE #61413: [BUG] Use NET instead of SC for service management in salt-cloud on Windows
**Description**

The portion of salt-cloud that restarts the minion service after laying down new config is using something similar to the following to restart the minion service.

```
sc salt-minion stop
time.sleep(5)
sc salt-minion start
```

The `SC` command runs asynchronously. The return code is that the command was issued successfully, not that the service actually stopped or started. The 5 second sleep assumes enough time has passed that the service has stopped. This presents a race condition where sometimes the service takes more than 5 seconds to stop, so you don't always see the error.

The `NET` command should be used instead. It runs synchronously and the return code is the success or failure of the start/stop action. This negates the need for the sleep command and avoids the race condition.

Details can be found here: From: https://jira.eng.vmware.com/browse/VRAE-8756

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Uses `net` instead of `sc` to restart the salt-minion service on Windows. The `net` command will not return until it completes, where the `sc` command just fires the `stop` or `start` and returns. That's why there was a 5 second sleep in between the `stop` and `start` commands. Sometimes the stop command takes longer than 5 seconds and the start command then fails because the service is still shutting down. Using `net` avoids this race condition and removes the need for the sleep command.

### What issues does this PR fix or reference?
Fixes: https://jira.eng.vmware.com/browse/VRAE-8756

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/utils/cloud.py [hunk 1]
```diff
@@ -1388,21 +1388,19 @@ def deploy_windows(
                 salt.utils.smb.delete_directory("salttemp", "C$", conn=smb_conn)
         # Shell out to psexec to ensure salt-minion service started
         if use_winrm:
-            winrm_cmd(winrm_session, "sc", ["stop", "salt-minion"])
+            winrm_cmd(winrm_session, "net", ["stop", "salt-minion"])
```
Comment (Ch3LL) [inline salt/utils/cloud.py:1391]: **question:** : Is `net` available by default on all windows platforms we support?

**Review by Ch3LL — CHANGES_REQUESTED** [review]
This requires a changelog

File: salt/utils/cloud.py (hunk 1 above)
Comment (twangboy) [reply salt/utils/cloud.py:1391]: yes

**Comment by twangboy** [comment]
@Ch3LL Added changelog

**Review by Ch3LL — APPROVED** [review]

**Review by garethgreenaway — APPROVED** [review]

**Review by MKLeb — APPROVED** [review]

**Review by waynew — APPROVED** [review]

**Pull request merged**